src/main.cpp

```diff
@@ -2382,7 +2382,7 @@ bool ConnectBlock(const CBlock &block,
     bool fScriptChecks = true;
     if (pindexBestHeader)
         fScriptChecks = !fCheckpointsEnabled || block.nTime > timeBarrier ||
-                        pindex->nHeight > pindexBestHeader->nHeight - (144 * checkScriptDays.value);
```

Comment:
Pedantic, but might affect regtest:
doing it this way means you could roll under 0 to a large number and then cast that to an integer.  If datatypes match this cast would = the same negative number.  But if the cast to int also changes the size of the data, then other things could happen.

So it would be better to cast before subtracting:
 (((int)pindexBestHeader->nHeight) - (144 * checkScriptDays.value));